### ISSUE TITLE
Enable building allConfigurations for corefx in source-build

### ISSUE THREAD

**Issue by weshaggard** [dotnet/source-build#210]
We need to build all packages not just .NET Core app packages when we are in offline source-build mode. However the allconfigurations leg has never worked correctly on non-windows so we need to iterate on it in order to enable.

We need to essentially undo https://github.com/dotnet/source-build/pull/195/commits/617ebe076b251f3691ad77add56eae5cca872f1a.

**Comment by weshaggard** [comment]
When we enable this we need to take into account https://github.com/dotnet/source-build/issues/184

**Comment by weshaggard** [comment]
@bartonjs this is likely something we will need to do at some point to fully build the packages people depend on upstack.

**Comment by dagood** [comment]
Status:

I worked on this for a while. I'm able to make CoreFX build AllConfigurations on linux after a few changes. WIP branch: https://github.com/dotnet/source-build/compare/master...dagood:corefx-all

Initially I tried to use a single command to build `-allconfigurations` and the vertical for the current platform within the same build context. That has complications (https://github.com/dotnet/corefx/pull/34332#pullrequestreview-189954566) so I abandoned it and instead first execute a build command for `-allconfigurations`, then another build command for the vertical.

The build passes with AllConfigurations included, but it brings in a ton of new prebuilts: https://github.com/dotnet/source-build/commit/68c057804fe788a1db8b605bc8602af3e79fb48c. Main categories are:

1. Old versions of CoreFX packages to harvest from.
1. CoreCLR builds for other platforms (e.g. to compile against the build Windows assemblies on Linux).
1. A CoreCLR build for the current platform. When I tried to use the source-built CoreCLR, it failed:
   * The locally built version doesn't necessarily line up with a Microsoft-built version for the other platforms.
   * Building AllConfigurations might make the coupling between CoreCLR and CoreFX more brittle than before with only one vertical. I don't know for sure if this caused the errors I was seeing, though.
1. A little extra build tooling. (I haven't investigated to see if we can source-build it.)

I experimented with forcing them to be treated as reference packages, but hit problems with resource files in the ILDasm/ILAsm round trip.
